x-pack/plugins/security/server/authentication/tokens.ts

```diff
@@ -70,8 +70,9 @@ export class Tokens {
 
       return {
         accessToken,
+        // @ts-expect-error @elastic/elasticsearch `refresh_token` might be undefined
```

Comment:
@elastic/kibana-security should we account for this potential `undefined` possibility in our code?

Reply:
Did the implementation change or was the types just updated to reflect a potential `undefined` value that could always have been there?

Reply:
AFAIK, only the types returned by ES changed.

Reply:
> // @ts-expect-error @elastic/elasticsearch `refresh_token` might be undefined

The `getToken` API accepts different grant types, and some grant types might not assume a refresh token (e.g. `grant_type === client_credentials`).

In this particular case, we use `refresh_token` grant type that will force `getToken` API to always return defined refresh token.

@afharo, unless we have a way to tweak `getToken` type definition to specify different response types depending on the `grant_type`, I'd rather use a non-null assertion + comment instead of `@ts-expect-error`. This way we won't miss other type issues. What do you think?
```ts
        // We can safely use a non-null assertion for the refresh token since `refresh_token` grant type guarantees that
        // getToken API will always return a new refresh token, unlike some other grant types (e.g. client_credentials).
        refreshToken: refreshToken!,
```

Reply:
@azasypkin that makes a lot of sense! I'll change that!